### PR TITLE
chore(deps): bump alloy

### PR DESCRIPTION
**Pull request by klkvr**
Bump alloy to latest commit. This currently uses several patches:
- revm: https://github.com/bluealloy/revm/pull/1695
- revm-inspectors: https://github.com/paradigmxyz/revm-inspectors/pull/184
- op-alloy: https://github.com/alloy-rs/op-alloy/pull/45

Main changes:
- `alloy_rpc_types_eth::Transaction` no more has `other` field, so I've updated code that relies on it to use `WithOtherFields<Transaction>` instead. Also updated `NetworkTypes` value to `AnyNetwork`.
- Updated `Compact` impl for `Bytecode`. It was incorrect for EOF and needed an impl for EIP7702 case

There's now `deposit_receipt_version` field on `OptimismTransactionFields`. I've left it as `None` for now, not sure how we should populate it

### CODE REVIEW

**Comment by mattsse** [comment]
> There's now deposit_receipt_version field on OptimismTransactionFields. I've left it as None for now, not sure how we should populate it

yeah we can leave this None for now, we need to add support for this separately

File: crates/e2e-test-utils/src/node.rs [hunk 1]
```diff
@@ -94,7 +95,7 @@ where
             From<<Node::Engine as PayloadTypes>::BuiltPayload> + PayloadEnvelopeExt,
         AddOns::EthApi: EthApiSpec + EthTransactions + TraceExt,
         <AddOns::EthApi as EthApiTypes>::NetworkTypes:
-            Network<TransactionResponse = alloy_rpc_types::Transaction>,
+            Network<TransactionResponse = WithOtherFields<alloy_rpc_types::Transaction>>,
```
Comment (mattsse) [inline crates/e2e-test-utils/src/node.rs:98]: @emhane how feasible do you think using OpNetwork is after this pr?

File: crates/primitives-traits/src/account.rs [hunk 2]
```diff
@@ -66,9 +66,14 @@ impl Compact for Bytecode {
...
-        let bytecode = &self.0.bytecode()[..];
+        let bytecode = match &self.0 {
+            RevmBytecode::LegacyRaw(bytes) => bytes,
+            RevmBytecode::LegacyAnalyzed(analyzed) => analyzed.bytecode(),
+            RevmBytecode::Eof(eof) => eof.raw(),
+            RevmBytecode::Eip7702(eip7702) => eip7702.raw(),
+        };
         buf.put_u32(bytecode.len() as u32);
-        buf.put_slice(bytecode);
+        buf.put_slice(bytecode.as_ref());
```
Comment (mattsse) [inline crates/primitives-traits/src/account.rs:76]: fyi @onbjerg

Comment (onbjerg) [reply crates/primitives-traits/src/account.rs:76]: looks correct

Comment (mattsse) [reply crates/primitives-traits/src/account.rs:76]: this is required for the codec I believe

**Comment by klkvr** [comment]
lint job is failing because `reth-primitives` only needs `op-alloy-rpc-types` if both `optimism` and `alloy-compat` features are enabled, and compiling it with only one of them causes unused dependency warning

should I just disable this check for that case?

**Comment by mattsse** [comment]
I think we can cheat here by doing

> or add `use op_alloy_rpc_types as _;` to the crate root

**Comment by onbjerg** [comment]
revm released

**Review by onbjerg — APPROVED** [review]
lgtm

**Review by onbjerg — APPROVED** [review]

**Review by mattsse — APPROVED** [review]

**Pull request merged**